src/Selenium2Driver.php

```diff
@@ -101,9 +101,12 @@ protected function setBrowserName($browserName = 'firefox')
     public function setDesiredCapabilities($desiredCapabilities = null)
     {
         if (null === $desiredCapabilities) {
-            $desiredCapabilities = self::getDefaultCapabilities();
+            $desiredCapabilities = array();
         }
 
+        // Join $desiredCapabilities with defaultCapabilities
+        $desiredCapabilities = $desiredCapabilities + self::getDefaultCapabilities();
```

Comment:
``+`` does the same than ``array_replace`` (except it works on PHP 5.2 while ``array_replace`` is 5.3+). So it does not replace recursively. But the default capabilities don't include any nested one anyway.

@andrerom I prefer using ``array_replace`` though. It makes the code more readable.